bindings/profilers/heap.cc

```diff
@@ -17,12 +17,93 @@
 #include "heap.hh"
 
 #include <memory>
+#include <vector>
 
 #include <node.h>
 #include <v8-profiler.h>
 
 using namespace v8;
 
+static Nan::Callback *cb;
+static bool s_dump_allocation_profile;
+static uv_work_t req;
+static std::unique_ptr<v8::AllocationProfile> s_allocation_profile;
+static uv_async_t s_async;
+
+static void dumpAllocationProfile(FILE* file, AllocationProfile::Node* node, std::string & cur_stack) {
+  auto len = cur_stack.size();
+  auto* isolate = v8::Isolate::GetCurrent();
+  char buf[256];
+  Nan::Utf8String name(node->name);
+  Nan::Utf8String script_name(node->script_name);
+  
+  snprintf(buf, sizeof(buf), "%s%s:%s:%d", cur_stack.empty() ? "" : ";", 
+           script_name.length() ? *script_name : "_",
+           name.length() ? *name : "(anonymous)",
+           node->line_number);
+  cur_stack += buf;
+  for(auto& allocation: node->allocations) {
+    fprintf(file, "%s %u %zu\n", cur_stack.c_str(), allocation.count, allocation.count * allocation.size);
```

Comment:
We'll have to check if there are potential limitations with how logs can be gathered, or how they should be formatted.